web/ai_service.py

```
@@ -0,0 +1,267 @@
+            if response and response.text:
+                return response.text
+            return None
+            
+        except KeyboardInterrupt:
+            print("\nGemini API request interrupted")
+            return None
+        except Exception as e:
+            print(f"Gemini API error: {str(e)}")
+            return None
+
+    def _get_openai_response(self, prompt: str, context: Optional[Dict[str, Any]] = None) -> Optional[str]:
```

Ruff (0.8.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)